bundles/binding/org.openhab.binding.isy/src/main/java/org/openhab/binding/isy/ISYGenericBindingProvider.java

```diff
@@ -0,0 +1,200 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.isy;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.openhab.binding.isy.internal.ISYControl;
+import org.openhab.binding.isy.internal.ISYNodeType;
+import org.openhab.core.binding.BindingConfig;
+import org.openhab.core.items.GroupItem;
+import org.openhab.core.items.Item;
+import org.openhab.core.library.items.ContactItem;
+import org.openhab.core.library.items.NumberItem;
+import org.openhab.core.library.items.StringItem;
+import org.openhab.core.library.items.SwitchItem;
+import org.openhab.core.library.items.DimmerItem;
+import org.openhab.core.library.items.DateTimeItem;
+import org.openhab.model.item.binding.AbstractGenericBindingProvider;
+import org.openhab.model.item.binding.BindingConfigParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is responsible for parsing the binding configuration.
+ *
+ * @author Tim Diekmann
+ * @author Jon Bullen
+ * @since 1.7.0
+ * 
+ */
+public class ISYGenericBindingProvider extends AbstractGenericBindingProvider
+		implements ISYBindingProvider {
+	private final Logger logger = LoggerFactory.getLogger(getClass());
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public String getBindingType() {
+		return "isy";
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	public void validateItemType(final Item item, final String bindingConfig)
+			throws BindingConfigParseException {
+		if (!(item instanceof ContactItem || item instanceof SwitchItem 
+				|| item instanceof NumberItem || item instanceof StringItem
+				|| item instanceof DateTimeItem)) {
+			throw new BindingConfigParseException(
+					"item '"
+							+ item.getName()
+							+ "' is of type '"
+							+ item.getClass().getSimpleName()
+							+ "', only Switch, Contact, String, Number and DateTime Items are supported yet - please check your *.items configuration");
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void processBindingConfiguration(final String context, final Item item, final String bindingConfig)
+			throws BindingConfigParseException {
+		super.processBindingConfiguration(context, item, bindingConfig);
```

Comment:
move to bottom of method so bcpe can be thrown to avoid holding onto invalid binding config strings.